### PR TITLE
feat(client): log Data Proxy usage, client type and host

### LINKED ISSUE #14319: Prisma Client: Log Data Proxy usage explicitly
When you use Prisma Client, it is not obvious if you are using a direct connection or if you are talking to a Data Proxy instance. This leads to questions from users, as they are not sure generating a Data Proxy Client worked and if it is successfully using the Data Proxy connection string.

We should find a way to log the fact that they are using a Data Proxy Client so it is easier to confirm everything is working as expected.

Internal discussion about user need: https://prisma-company.slack.com/archives/C02GF9J74SH/p1657611693346579
Internal discussion about hostname of Data Proxy logging: https://prisma-company.slack.com/archives/C02C34KRHBQ/p1657878477219309

### PR DESCRIPTION
**Pull request by aqrln**
Closes https://github.com/prisma/prisma/issues/14319

### CODE REVIEW

File: packages/client/src/runtime/getPrismaClient.ts [hunk 1]
```diff
@@ -466,7 +466,12 @@ export function getPrismaClient(config: GetPrismaClientConfig) {
         }
 
         debug(`clientVersion: ${config.clientVersion}`)
-        debug(`clientEngineType: ${this._clientEngineType}`)
+        debug(`clientEngineType: ${this._dataProxy ? 'dataproxy' : this._clientEngineType}`)
+
+        if (this._dataProxy) {
+          const runtime = NODE_CLIENT ? 'Node.js' : 'edge'
+          debug(`using Data Proxy with ${runtime} client`)
```
Comment (millsp) [inline packages/client/src/runtime/getPrismaClient.ts:473]:
```suggestion
          debug(`using Data Proxy with ${runtime} runtime`)
```

File: packages/client/src/runtime/getPrismaClient.ts [hunk 2]
```diff
@@ -466,7 +466,12 @@ export function getPrismaClient(config: GetPrismaClientConfig) {
         }
 
         debug(`clientVersion: ${config.clientVersion}`)
```
Comment (millsp) [inline packages/client/src/runtime/getPrismaClient.ts:468]:
```suggestion
        debug('clientVersion', config.clientVersion)
```

File: packages/client/src/runtime/getPrismaClient.ts [hunk 3]
```diff
@@ -466,7 +466,12 @@ export function getPrismaClient(config: GetPrismaClientConfig) {
...
-        debug(`clientEngineType: ${this._clientEngineType}`)
+        debug(`clientEngineType: ${this._dataProxy ? 'dataproxy' : this._clientEngineType}`)
```
Comment (millsp) [inline packages/client/src/runtime/getPrismaClient.ts:469]:
```suggestion
        debug('clientEngineType', this._dataProxy ? 'dataproxy' : this._clientEngineType)
```

File: packages/engine-core/src/data-proxy/DataProxyEngine.ts [hunk 4]
```diff
@@ -51,6 +54,8 @@ export class DataProxyEngine extends Engine {
     this.headers = { Authorization: `Bearer ${apiKey}` }
     this.host = host
 
+    debug(`Data Proxy host: ${this.host}`)
```
Comment (millsp) [inline packages/engine-core/src/data-proxy/DataProxyEngine.ts:57]:
```suggestion
    debug('host', this.host)
```

**Review by janpio — APPROVED** [review]
👍 

Wouldn't it be nice to have snapshot tests for this now to see it in action :D

File: packages/engine-core/src/data-proxy/DataProxyEngine.ts [hunk 5]
```diff
@@ -51,6 +54,8 @@ export class DataProxyEngine extends Engine {
     this.headers = { Authorization: `Bearer ${apiKey}` }
     this.host = host
 
+    debug('host', this.host)
```
[the remark this answers is not in the document]
Comment (aqrln) [reply packages/engine-core/src/data-proxy/DataProxyEngine.ts:57]: It's actually already logged in other places:

https://github.com/prisma/prisma/blob/8a414325f457b5d1023a821bd31fe118bad308a8/packages/engine-core/src/data-proxy/DataProxyEngine.ts#L146-L148

**Review by millsp — APPROVED** [review]

File: packages/engine-core/src/data-proxy/DataProxyEngine.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (aqrln) [reply packages/engine-core/src/data-proxy/DataProxyEngine.ts:57]: > Might ask Scale team if they consider the host a secret or if they are fine with that appearing in Prisma Client logs

Actually you already asked them three weeks ago 😄
According to [this thread](https://prisma-company.slack.com/archives/C02C34KRHBQ/p1657878477219309), it's not a secret and they are fine with it.

[the remark this answers is not in the document]
Comment (janpio) [reply packages/engine-core/src/data-proxy/DataProxyEngine.ts:57]: Ok, then let's ignore that here and create a new "Future work" note that we should check that. Logging the full URLs does not sounds like the smartest idea either honestly.

[the remark this answers is not in the document]
Comment (janpio) [reply packages/engine-core/src/data-proxy/DataProxyEngine.ts:57]: I am so smart :D

Ok then :p

**Pull request merged**